components/script/dom/htmlelement.rs

```diff
@@ -817,6 +817,19 @@ impl HTMLElement {
         }
     }
 
+    /// <https://html.spec.whatwg.org/multipage/#the-body-element-2>
+    pub(crate) fn is_body_element(&self) -> bool {
+        let self_node = self.upcast::<Node>();
+        self_node.GetParentNode().is_some_and(|parent| {
+            let parent_node = parent.upcast::<Node>();
+            self_node.is::<HTMLBodyElement>() &&
```

Comment:
This should also check `HTMLFrameSetElement`

Reply:
No, that one is misleading. It's actually a method to check if event target is body or frameset.